### PR TITLE
Memfault sdk esp component article

### PR DESCRIPTION
**Pull request by noahp**

### CODE REVIEW

File: _drafts/memfault-esp-component.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+## What are we talking about here
+
+Ok, so full disclosure- your author is employed by Memfault, and this post is
+going to be talking a LOT about our
+[Memfault Firmware SDK](https://github.com/memfault/memfault-firmware-sdk)! In
+spite of that, I hope it's an iteresting walkthrough of this specific problem we
```
Comment (noahp) [inline _drafts/memfault-esp-component.md:26]:
```suggestion
spite of that, I hope it's an interesting walkthrough of this specific problem we
```

File: _drafts/memfault-esp-component.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+ESP Component Registry-enabled components have the follow properties:
+
+- Root-level `idf_component.yml` manifest
+- Root-level `CMakeLists.txt` and `Kconfig` files
+- The enclosing folder name must _exactly_ match the component name
+
+We have essentially 2 options when it comes to making our SDK an ESP-IDF
+component:
+
+1. **Move the SDK into a new folder that matches the component name**:
```
Comment (noahp) [inline _drafts/memfault-esp-component.md:135]: This is very inaccurate- it's more like "reorganize our SDK to put the esp port at the root, and the core files under src etc"

File: _drafts/memfault-esp-component.md [hunk 3]
```diff
@@ -0,0 +1,368 @@
...
+
+- Confirm the component builds every time we push a change to the internal
+  development repo for the SDK
+- A nightly check that the latest published component on the registry builds
+  successfully
+- A second nightly check that confirms the component builds successfully with
+  the latest ESP-IDF master branch, in case there's upcoming tooling or build
+  changes that we need to be aware of
+
+## Summary
```
Comment (noahp) [inline _drafts/memfault-esp-component.md:333]: I need to link to the final component!

**Review by gminn — COMMENTED** [review]
A couple comments, and I'll fix things up once I get your thoughts!

File: _drafts/memfault-esp-component.md [hunk 4]
```diff
@@ -0,0 +1,368 @@
...
+  --archive dist/memfault-firmware-sdk_1.11.4-rc1.tgz \
+  --namespace memfault \
+  --name memfault-firmware-sdk
+```
+
+> Note: the example command above is uploading the component to the `memfault`
+> namespace. To register a new namespace, you'll need to create an account on
+> the ESP Component Registry website- the namespace will be set to the user
+> account name. Espressif supports custom namespaces, see
+> [docs here](https://docs.espressif.com/projects/idf-component-manager/en/latest/guides/packaging_components.html#authentication).
```
[the remark this answers is not in the document]
Comment (noahp) [reply _drafts/memfault-esp-component.md:278]: Ah yeah, good call- the espressif doc is a little confusing, it's more like "if you need a custom namespace, contact us and we can create it for you"

[the remark this answers is not in the document]
Comment (noahp) [reply _drafts/memfault-esp-component.md:278]: Updated!

**Review by gminn — APPROVED** [review]

**Pull request merged**